policy-controller/grpc/src/lib.rs

```diff
@@ -320,5 +342,198 @@ fn to_authz(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Get.into(),
+                        )
+                    }
+                    HttpMethod::POST => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Post.into(),
+                        )
+                    }
+                    HttpMethod::PUT => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Put.into(),
+                        )
+                    }
+                    HttpMethod::DELETE => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Delete.into(),
+                        )
+                    }
+                    HttpMethod::PATCH => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Patch.into(),
+                        )
+                    }
+                    HttpMethod::HEAD => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Head.into(),
+                        )
+                    }
+                    HttpMethod::OPTIONS => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Options.into(),
+                        )
+                    }
+                    HttpMethod::TRACE => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Registered(
+                            linkerd2_proxy_api::http_types::http_method::Registered::Trace.into(),
+                        )
+                    }
+                    HttpMethod::Unregistered(method) => {
+                        linkerd2_proxy_api::http_types::http_method::Type::Unregistered(
+                            method.to_owned(),
+                        )
+                    }
+                };
+                linkerd2_proxy_api::http_types::HttpMethod { r#type: Some(typ) }
+            });
+
+            linkerd2_proxy_api::http_route::HttpRouteMatch {
+                headers,
+                path,
+                query_params,
+                method,
+            }
+        })
+        .collect();
+
+    let rules = vec![proto::http_route::Rule {
+        matches,
+        filters: Vec::new(),
+    }];
+
+    proto::HttpRoute {
+        metadata: Some(metadata),
+        hosts,
+        authorizations,
+        rules,
```

Comment:
I'd be inclined to break this into smaller converter functions for each type, for readability.